replications/internal/queue_management.go

```diff
@@ -368,6 +393,30 @@ func (qm *durableQueueManager) EnqueueData(replicationID platform.ID, data []byt
 	return nil
 }
 
+func getBatch(data []byte, batchSize int) ([]byte, []byte) {
```

Comment:
The data from the disk queue is going to be gzipped, see: https://github.com/influxdata/influxdb/blob/a74e05177c7dc92b75ffe70833d0790f2a923b46/replications/service.go#L414-L475 ... so we won't be able to parse the line protocol like this, and we definitely won't want to decompress it for this check (resource intensive).

I think we'll need to put the batching logic into `WritePoints` of the replications "service" struct: As the points are being processed to add to the queue, sum their pre-gzip lengths (sizes) until the threshold is reached, and if it is, write that batch to the queue and start a new one. The queue reading side won't have to worry about batch sizes at all then, which will be good since it really can't know the uncompressed size of the LP without doing some really costly decompression which we will want to avoid.

Reply:
I've made an attempt at refactoring this. We will still need a test in `service_test.go` that checks for proper batching though - I tried writing a test similar to `TestWritePoints()`, but with much more data (greater than the batch size limit), but wasn't able to get the test working. I think it might be good to get your thoughts on the `WritePoints()` function itself before spending more time stuck on the test.